### ISSUE TITLE
Implement `GetRecordsConfig.DoWithContext`

### ISSUE THREAD

**Issue by michalb-goflink** [mehanizm/airtable#19]
Hello, thanks for the nice library :)

We use code like:

```go
records, err := table.GetRecords().
		FromView(viewID).
		WithOffset(offset).
		Do()
```

and we have issue with `Do()` that calls `Table.GetRecordsWithParams` which calls `Table.GetRecordsWithParamsContext` using default context `context.Background()`. We would like to have option to pass custom context instance.

The request I have is to implement `DoWithContext(context.Context)` that does just that. I could not find a way to achieve passing context, as in the end `GetRecordsConfig.params` are private

**Comment by mehanizm** [comment]
@michalb-goflink , thank you
Fixed v0.3.2+

**Comment by michalb-goflink** [comment]
@mehanizm nice, thanks
